redisvl/schema/schema.py

```diff
@@ -0,0 +1,306 @@
+import os
+import yaml
+
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+from pydantic import BaseModel, ValidationError
+
+from redisvl.schema.fields import (
+    TagFieldSchema,
+    TextFieldSchema,
+    NumericFieldSchema,
+    FlatVectorFieldSchema,
+    HNSWVectorFieldSchema,
+    GeoFieldSchema
+)
+
+from redisvl.utils.utils import (
+    convert_bytes,
+    make_dict,
+)
+
+
+class StorageType(Enum):
+    HASH = "hash"
+    JSON = "json"
+
+
+class IndexModel(BaseModel):
```

Comment:
At this point I think we just collapse ``IndexModel`` and ``FieldsModel`` into the Schema object. No reason to nest them from what I see here.

Reply:
I think the benefit to keeping them here is for pydantic validation since Schema is not a pydantic object. Each has separate conditions and checks. Also makes it cleaner than accessing args from a dictionary like `self.index['prefix']`. Obviously a few routes to handle this:

1. remove the model classes and have everything be a first class citizen within the Schema object namespace `schema.name` schema.prefix` `schema.tags` etc

Reply:
I think there are some advantages to keeping it. Will push a new version of code and lets review together so we can discuss